code/io/joy.h

```diff
@@ -17,7 +17,7 @@
 #include "SDL_joystick.h"
 
 // z64: Moved up here for compatibility. Bye bye, organization!
-const int JOY_NUM_BUTTONS = 32;
+const int JOY_NUM_BUTTONS = 128;	// Max number of buttons FSO can handle.  OS max may differ.
```

Comment:
What are the ramifications of bumping this limit, as far as FSO data structures go?  A cursory search shows no arrays using `JOY_NUM_BUTTONS`, but a more extensive search shows `Joy_button_text` uses the old limit and would produce out-of-range errors using the new limit.

(As a consequence of this, it would be good to explicitly specify `JOY_NUM_BUTTONS` in the declaration of `Joy_button_text_english[]` and similar arrays.)

Reply:
as far as I can remember, none of the controls structures would be affected by the joy button bump.  Localization code is probably the only thorn to the bump which may be the reason why I had not done the bump back in the Controls code overhaul.

I'll uh, try poking around today to see if I can find all instances where this may be an issue.

Reply:
Ok.  This has some side effects with presets, so I'm reverting the button number back down to 32 and will defer this feature to another PR.